src/main/scala/it/unibo/authsim/library/dsl/attack/builders/BruteForceAttackBuilder.scala

```diff
@@ -81,8 +82,8 @@ private class BruteForceAttack(private val target: Proxy, private val hashFuncti
       val hashedPassword = hashFunction.hash(nextPasswordString)
       localStatistics = localStatistics + new Statistics(hashedPassword == targetUser.password match {
         case true => Set(new User {
-          override def username: String = targetUser.username
-          override def password: String = nextPasswordString
+          override val username: String = targetUser.username
+          override val password: String = nextPasswordString
```

Comment:
Here you can use the companion object to generate `User`s, as it would avoid the anonymous extension (and improve readability).

Reply:
Wait, is this not one of your test class? I do not understand what do you mean `User` trait has its own companion object.

Reply:
This is an implementation class: the anonymous extension was made when the `User` trait had no companion object. Now that `User` has a companion object which provides an `apply` method to create `User`s, in this point of the code it can be used.

Now it can be changed to:
```
case true => Set(User(targetUSer.username, nextPasswordString))
```
with the parenthesis correctly matched (after the snippet there is another closed bracket `}`, after the substitution it must be deleted)

Reply:
Ah ok there are no errors, you just want me to change your class implementation in reason of the _new implementation_ of `User`, now I got it.

Reply:
Yes, if you don't mind